File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/03-additional/12.theSongOfTheWheels.js
function theSongOfTheWheels(input) {

    let controlValue = Number(input[0]);

    let combinations = '';
    let counter = 0;
    let pass = '';

    for (let a = 1; a <= 9; a++) {
        for (let b = 1; b <= 9; b++) {
            for (let c = 1; c <= 9; c++) {
                for (let d = 1; d <= 9; d++) {
                    if (a < b && c > d && a * b + c * d == controlValue) {
                        combinations += `${a}${b}${c}${d} `;
                        counter++;
                        if (counter == 4) {
                            pass += `${a}${b}${c}${d}`;
                        }
                    }
                }
            }
        }
    }

    if (combinations.length > 0) {
        console.log(combinations);
    }
    if (pass.length > 0) {
        console.log(`Password: ${pass}`);
    } else {
        console.log('No!');
    }
}
theSongOfTheWheels(['55']);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/03.additional/12.the_song_of_the_wheels.py
control_value = int(input())

combinations = ''
counter = 0
password = ''

for a in range(1, 10):
    for b in range(1, 10):
        for c in range(1, 10):
            for d in range(1, 10):
                if a < b and c > d and a * b + c * d == control_value:
                    combinations += f'{a}{b}{c}{d} '
                    counter += 1
                    if counter == 4:
                        password += f'{a}{b}{c}{d}'

if len(combinations) > 0:
    print(combinations)
if len(password) > 0:
    print(f'Password: {password}')
else:
    print('No!')
